src/main/java/gregtech/client/renderer/texture/custom/QuantumStorageRenderer.java

```diff
@@ -0,0 +1,147 @@
+package gregtech.client.renderer.texture.custom;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.texture.TextureUtils;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import codechicken.lib.vec.Rotation;
+import codechicken.lib.vec.Transformation;
+import gregtech.api.GTValues;
+import gregtech.api.gui.resources.TextTexture;
+import gregtech.api.util.TextFormattingUtil;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.utils.RenderUtil;
+import gregtech.common.metatileentities.storage.MetaTileEntityQuantumChest;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class QuantumStorageRenderer implements TextureUtils.IIconRegister {
+    private static final Cuboid6 upBox = new Cuboid6(0 / 16.0, 14 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 downBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 2 / 16.0, 16 / 16.0);
+    private static final Cuboid6 westBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 2 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 eastBox = new Cuboid6(14 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 southBox = new Cuboid6(0 / 16.0, 0 / 16.0, 14 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 northBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 2 / 16.0);
+    private static final Cuboid6 glassBox = new Cuboid6(1 / 16.0, 1 / 16.0, 1 / 16.0, 15 / 16.0, 15 / 16.0, 15 / 16.0);
+    private static final Cuboid6 fluidBox = new Cuboid6(2 / 16.0, 2 / 16.0, 2 / 16.0, 14 / 16.0, 14 / 16.0, 14 / 16.0);
+
+
+    private static Map<EnumFacing, Cuboid6> boxFacingMap = new HashMap<>();
+
+    @SideOnly(Side.CLIENT)
+    private TextureAtlasSprite[] textures;
```

Comment:
I think this can be changed to `private static TextureAtlasSprite texture`, and only store the unique texture for the renderer instead. The other base casing texture can be retrieved from the rendered meta tile entity. See my comment about that in `renderMachine`.